internal/command/launch/launch.go

```diff
@@ -17,158 +17,158 @@ import (
+	"github.com/superfly/flyctl/internal/command"
+	"github.com/superfly/flyctl/internal/command/apps"
+	"github.com/superfly/flyctl/internal/command/deploy"
+	"github.com/superfly/flyctl/internal/command/postgres"
 	"github.com/superfly/flyctl/internal/filemu"
+	"github.com/superfly/flyctl/internal/flag"
+	"github.com/superfly/flyctl/internal/prompt"
+	"github.com/superfly/flyctl/iostreams"
 	"github.com/superfly/flyctl/scanner"
 	"github.com/superfly/flyctl/terminal"
 	"github.com/superfly/graphql"
 )
 
-func newLaunchCommand(client *client.Client) *Command {
-	launchStrings := docstrings.Get("launch")
-	launchCmd := BuildCommandKS(nil, runLaunch, launchStrings, client, requireSession)
-	launchCmd.Args = cobra.NoArgs
-	launchCmd.AddStringFlag(StringFlagOpts{
-		Name:        "path",
-		Description: `Path to app code and where a fly.toml file will be saved`,
-		Default:     ".",
-	},
+func New() (cmd *cobra.Command) {
+	const (
+		long  = `Create and configure a new app from source code or a Docker image.`
+		short = long
 	)
-	launchCmd.AddStringFlag(StringFlagOpts{
-		Name:        "org",
-		Description: `Organization that will own the app (use org slug here; see "fly orgs list")`,
-	})
-	launchCmd.AddStringFlag(StringFlagOpts{
-		Name:        "name",
-		Description: "Name of the new app",
-	})
-	launchCmd.AddStringFlag(StringFlagOpts{
-		Name:        "region",
-		Description: "Region to launch the new app in",
-	})
-	launchCmd.AddStringFlag(StringFlagOpts{
-		Name:        "image",
-		Description: "Image to launch",
-	})
-	launchCmd.AddBoolFlag(BoolFlagOpts{
-		Name:        "now",
-		Description: "Deploy now without confirmation",
-		Default:     false,
-	})
-	launchCmd.AddBoolFlag(BoolFlagOpts{
-		Name:        "no-deploy",
-		Description: "Do not prompt for deployment",
-		Default:     false,
-	})
-	launchCmd.AddBoolFlag(BoolFlagOpts{
-		Name:        "generate-name",
-		Description: "Always generate a name for the app",
-		Default:     false,
-	})
-	launchCmd.AddStringFlag(StringFlagOpts{
-		Name:        "dockerfile",
-		Description: "Path to a Dockerfile. Defaults to the Dockerfile in the working directory",
-	})
-	launchCmd.AddBoolFlag(BoolFlagOpts{
-		Name:        "copy-config",
-		Description: "Use the configuration file if present without prompting",
-		Default:     false,
-	})
-	launchCmd.AddBoolFlag(BoolFlagOpts{
-		Name:        "remote-only",
-		Description: "Perform builds remotely without using the local docker daemon",
-		Default:     false,
-	})
-	launchCmd.AddBoolFlag(BoolFlagOpts{
-		Name:        "dockerignore-from-gitignore",
-		Description: "If a .dockerignore does not exist create one from .gitignore files",
-		Default:     false,
-	})
 
-	return launchCmd
-}
-
-func runLaunch(cmdCtx *cmdctx.CmdContext) error {
-	ctx := cmdCtx.Command.Context()
-
-	dir := cmdCtx.Config.GetString("path")
+	cmd = command.New("launch", short, long, run, command.RequireSession, command.LoadAppConfigIfPresent)
+
+	cmd.Hidden = true
```

Comment:
why is it hidden?

Reply:
Good catch. That was copypasted from the machines launch command.